File: 0209/solution.java
//https://leetcode.com/problems/minimum-size-subarray-sum/

class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int head = 0,sum=0,ans = Integer.MAX_VALUE;
        
        for(int i=0;i<n;i++){
            sum += nums[i];
            while(sum >= target){
                ans = Math.min(ans,i-head+1);
                sum -= nums[head];
                head += 1;
            }
        }
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}

File: 0209/solution.py
#https://leetcode.com/problems/minimum-size-subarray-sum/

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        if sum(nums) < target:
            return 0
        head = 0
        sum_l = 0
        ans = len(nums)
        for i in range(len(nums)):
            sum_l += nums[i]
            while sum_l >= target:
                ans = min(ans,i-head+1)
                sum_l -= nums[head]
                head += 1
        return ans